test/e2e/apimachinery/flowcontrol.go

```diff
@@ -313,7 +310,7 @@ func getPriorityLevelNominalConcurrency(c clientset.Interface, priorityLevelName
 
 // createFlowSchema creates a flow schema referring to a particular priority
 // level and matching the username provided.
-func createFlowSchema(f *framework.Framework, flowSchemaName string, matchingPrecedence int32, priorityLevelName string, matchingUsernames []string) (*flowcontrol.FlowSchema, func()) {
+func createFlowSchema(ctx context.Context, f *framework.Framework, flowSchemaName string, matchingPrecedence int32, priorityLevelName string, matchingUsernames []string) *flowcontrol.FlowSchema {
```

Comment:
`createPriorityLevel` has the same signature, however, you choose to use `ginkgo.DeferCleanup(` here instead of returning a `cleanup` function , I think that is better to be consistent, maybe avoid the ginkgo primitives by now?

Reply:
Using `ginkgo.DeferCleanup` was often the right solution for dealing with contexts during test cleanups. Not using it would have been more work and the outcome would have been more complicated. I had two choices:
- change the cleanup function to accept a context, which also means changing the caller
- eliminate the cleanup function

Reply:
but in some places the context is provided in the constructor, so the closure should inherit it , no?

Reply:
Each context from Ginko is only valid in the function that it is passed into. Using the context of the `BeforeEach` or `It` also for the cleanup operation is a bug. That should never be done because the API calls in the cleanup operation then abort immediately with "context cancelled" - one reason why it is important to check the error of those calls.